java/arcs/core/storage/StorageProxy.kt

```diff
@@ -112,13 +108,16 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperation, T>(
      */
     suspend fun getParticleView(): ValueAndVersion<T> = getParticleViewAsync().await()
 
-    fun getParticleViewAsync(): CompletableDeferred<ValueAndVersion<T>> {
+    suspend fun getParticleViewAsync(): CompletableDeferred<ValueAndVersion<T>> {
         log.debug { "Getting particle view" }
         val future = CompletableDeferred<ValueAndVersion<T>>()
-        mutex.withLock {
+        syncMutex.withLock {
             if (synchronized) {
```

Comment:
The TS code sends a sync request if it's not synced now: https://github.com/PolymerLabs/arcs/blob/04515598f48c1331bbec291ee5caba21a82b2549/src/runtime/storageNG/storage-proxy.ts#L148
I messed up in my original impl to not port that. 
I think we can cache the value of synced inside the mutex, release the mutex, and if we were unsynced, send the syncReq to storage at that time. That should ensure we can't deadlock.